others/mongodb-101.md

```diff
@@ -0,0 +1,233 @@
+# Week XX Lesson: MongoDB
+
+**What we will learn today?**
+
+* SQL vs NoSQL
+* Why MongoDB?
+* Mongo Shell Basic Commands
+* MongoDB administration
+   * Aggregation
+   * Replication
+   * Sharding
+* What's next?
+
+# Before we start
+
+Download and install MongoDb from
+https://www.mongodb.com/download-center#community . Follow the instructions for
+your platform (Windows, Linux or OS X).
+
+Open two instances of terminal. Run `mongod` in the first instance - this will start MongoDB.
+ 
+In the second instance, first download the sample data script:
+```curl https://gist.githubusercontent.com/agiamas/35b2b954cc942f95709273d3cb9d2cf3/raw/1a0b732594c5f5ae23f638c03519a98364f8b0ff/mongodb_data.js > mongo_data.js```
+
+then import the data we just downloaded using 
+```mongo < mongo_data.js```
+
+Finally, run `mongo` in the second terminal instance.
+
+
+# Lesson 1 - SQL and NoSQL
+ 
+SQL is a well known paradigm in data storage and retrieval, serving us for decades. 
+Our data was living in mainframes and being stored and processed in isolation. 
+The nature of Web and mobile has created a paradigm shift in the past decade. Data has exploded in volume, veracity and velocity.
+We have many times semi-structured data of varying quality and unpredictable volumes of them.
+  
+This has led to a new breed of databases, the NoSQL ones.
+- Document databases, e.g. MongoDB
+- Graph stores, e.g. Neo4J
+- key-value stores, e.g. Redis
+- Wide-column stores , e.g. Cassandra
+
+In this class we will examine Document databases and its most popular one, MongoDB.
+
+First of all, how does MongoDB compare with traditional SQL queries that we have learned in the past?
+http://s3.amazonaws.com/info-mongodb-com/sql_to_mongo.pdf
```

Comment:
Would prefer to include this in the repo if possible as S3 buckets have a tendency to disappear after a while :)